l10n_br_delivery_nfe/wizards/stock_invoice_onshipping.py

```diff
@@ -0,0 +1,141 @@
+# Copyright (C) 2024 Diego Paradeda - KMEE
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, fields, models
+
+
+class StockInvoiceOnshipping(models.TransientModel):
+    _inherit = "stock.invoice.onshipping"
+
+    vol_ids = fields.One2many(
+        string="Volume IDs Data",
+        comodel_name="stock.invoice.onshipping.vol",
+        inverse_name="invoice_wizard_id",
+    )
+
+    def _build_invoice_values_from_pickings(self, pickings):
+        invoice, values = super()._build_invoice_values_from_pickings(pickings)
+
+        # Volumes
+        values["nfe40_vol"] = [
+            (
+                0,
+                0,
+                self._convert_volumes_to_write(vol_id),
+            )
+            for vol_id in self.vol_ids
+        ]
+
+        return invoice, values
+
+    def _convert_volumes_to_write(self, data):
+        """Prepare wizard lines for writing to fiscal document."""
+        return {
+            "nfe40_qVol": data.nfe40_qVol,
+            "nfe40_esp": data.nfe40_esp,
+            "nfe40_marca": data.nfe40_marca,
+            "nfe40_pesoL": data.nfe40_pesoL,
+            "nfe40_pesoB": data.nfe40_pesoB,
+        }
+
+    def _get_picking(self):
```

Comment:
Existe um método _load_pickings https://github.com/OCA/account-invoicing/blob/14.0/stock_picking_invoicing/wizards/stock_invoice_onshipping.py#L254 no objeto que devolve os pickings com a diferença que tem um filtro para o invoice_state "2binvoiced" isso seria um problema para o seu caso de uso? Parece ser melhor usar o _load_pickings

Reply:
Posso incluir uma chamada do _load_pickings dentro do _get_picking. Substituir completamente acho que não seria uma boa pois parte importante do _get_picking é forçar retornar apenas o primeiro active_id caso tenha mais de um